source/setup.Rmd

```diff
@@ -1,282 +1,288 @@
+More information on installing and running Docker on Ubuntu, as well as troubleshooting tips, can be found
+[in the online Docker documetation](https://docs.docker.com/engine/install/ubuntu/).
+
+### MacOS
+
+To install Docker on MacOS, visit [the online Docker
+documentation](https://docs.docker.com/desktop/install/mac-install/), and
+download the `Docker.dmg` installation file that is appropriate for your
+computer (you need to know whether your computer has an Intel processor an
+Apple processor). Double-click the file to open the installer, then drag
+the Docker icon to the Applications folder.  Double-click Docker.app in the
+Applications folder to start Docker. In the installation window, use the
+recommended settings.
+
+Once Docker is installed and running, follow the instructions below in the Windows section
+for finding and running the virtual machine we have made available for this book.
+More information on installing and running
+Docker on MacOS, as well as troubleshooting tips, can be found [in the online
+Docker documetation](https://docs.docker.com/desktop/install/mac-install/).
+
+### Windows
+
+To install Docker on Windows, visit [the online Docker documentation](https://docs.docker.com/desktop/install/windows-install/),
+and download the `Docker Desktop Installer.exe` file. Double-click the file to open the installer
+and follow the instructions on the installation wizard, choosing **WSL-2** instead of **Hyper-V** when prompted.
+
+Occasionally, when you first run Docker on Windows, you will encounter an error message. Some common errors you may see:
+
+- If you need to update WSL, you can enter `cmd.exe` in the Start menu to run the command line. Type `wsl --update` to update WSL.
+- If the admin account on your computer is different to your user account, you must add the user to the "docker-users" group. 
+  Run Computer Management as an administrator and navigate to `Local Users` and `Groups -> Groups -> docker-users`. Right-click to 
+  add the user to the group. Log out and log back in for the changes to take effect.
+- If you need to enable virtualization, you will need to edit your BIOS. Restart your computer, and enter the BIOS using the hotkey
+  (usually Delete, Esc, and/or one of the F# keys). Look for an "Advanced" menu, and under your CPU settings, set the "Virtualization" option
+  to "enabled". Then save the changes and reboot your machine. If you are not familiar with BIOS editing, you may want to find an expert 
+  to help you with this, as editing the BIOS can be dangerous. Detailed instructions for doing this are beyond the scope of this book. 
+More information on installing and running
+Docker on Windows, as well as troubleshooting tips, can be found [in the online
+Docker documetation](https://docs.docker.com/desktop/install/windows-install/).
+
+
+Once Docker is running (in either Windows or MacOS), you need to download and run the virtual machine that we have made available
```

Comment:
Maybe this could have its own header to make it clearer that it belongs to windows and mac?

Reply:
yeah i thought about doing this, but was feeling lazy. Since you also pointed it out I'll make sure to do this (both for JLab desktop and Docker, which need a similar edit)

Reply:
done!